components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -178,6 +180,23 @@ public Response authorize(@Context HttpServletRequest request, @Context HttpServ
         SessionDataCacheEntry sessionDataCacheEntry = null;
 
         try {
+            if(clientId != null) {
+                OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
+                try {
+                    String appState = oAuthAppDAO.getConsumerAppState(clientId);
+                    if(!OAuthConstants.OauthAppStates.APP_STATE_ACTIVE.equalsIgnoreCase(appState)) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Oauth App is not in active state.");
+                        }
+                        OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse.SC_UNAUTHORIZED)
+                                .setError(OAuth2ErrorCodes.INVALID_CLIENT)
+                                .setErrorDescription("Oauth application is not in active state.").buildJSONMessage();
+                        return Response.status(oAuthResponse.getResponseStatus()).entity(oAuthResponse.getBody()).build();
+                    }
+                } catch (IdentityOAuthAdminException e) {
+                    throw new URISyntaxException("Error in getting oauth app state.", "Error in getting oauth app state.");
```

Comment:
When we throw a `URISyntaxException` what sort of a response will the client get ? Shouldn't we build an error response here ?